File: 3105-longest-strictly-increasing-or-strictly-decreasing-subarray/3105-longest-strictly-increasing-or-strictly-decreasing-subarray.cpp
class Solution {
public:
    int max(int a, int b) {
        return a > b ? a : b;
    }
    int longestMonotonicSubarray(vector<int>& nums) {
        int ans = 1;
        int inc = 1, dec = 1;
        int n = nums.size();
        for (int i = 0; i < n - 1; i++) {
            if (nums[i] < nums[i+1]) inc++;
            else inc = 1;
            if (nums[i] > nums[i+1]) dec++;
            else dec = 1;
            ans = max(ans, max(inc, dec));
        }
        return ans;
    }
};

File: LeetCode/3105-longest-strictly-increasing-or-strictly-decreasing-subarray/3105-longest-strictly-increasing-or-strictly-decreasing-subarray.py
class Solution:
    def longestMonotonicSubarray(self, nums: List[int]) -> int:
        ans = 1
        inc = dec = 1
        n = len(nums)
        
        for i in range(n - 1):
            if nums[i] < nums[i + 1]:
                inc += 1
            else:
                inc = 1
            
            if nums[i] > nums[i + 1]:
                dec += 1
            else:
                dec = 1
            
            ans = max(ans, inc, dec)
        
        return ans